tests/Serializer/SerializerTest.php

```diff
@@ -95,11 +96,45 @@ public function testNull(bool $serializeAllObjects): void
         $this->assertNull($result);
     }
 
+    public function testLongStringWithOverwrittenMessageLength(): void
+    {
+        $serializer = $this->createSerializer(new Options(['truncation_length' => 500]));
+
+        foreach ([100, 490, 499, 500, 501, 1000, 10000] as $length) {
+            $input = str_repeat('x', $length);
+            $result = $this->invokeSerialization($serializer, $input);
+
+            $this->assertInternalType('string', $result);
+            $this->assertLessThanOrEqual(500, \strlen($result));
+        }
+    }
+
+    public function testClippingUTF8Characters(): void
+    {
+        if (!\extension_loaded('mbstring')) {
+            $this->markTestSkipped('mbstring extension is not enabled.');
+        }
+
+        $serializer = $this->createSerializer(new Options(['truncation_length' => 19]));
+
+        $clipped = $this->invokeSerialization($serializer, 'Прекратите надеяться, что ваши пользователи будут сообщать об ошибках');
+
+        $this->assertSame('Прекратит {clipped}', $clipped);
+        $this->assertNotNull(json_encode($clipped));
+        $this->assertSame(JSON_ERROR_NONE, json_last_error());
+    }
+
     /**
+     * @param Options $options
+     *
      * @return Serializer
      */
-    protected function createSerializer(): AbstractSerializer
+    protected function createSerializer(Options $options = null): AbstractSerializer
```

Comment:
Since this is a `nullable` argument with a default value it should be `?Options $options = null`